### PR TITLE
Add goodparts to project

### PR DESCRIPTION
**Pull request by eliasmalik**
#### Changes
- Add goodparts as a dev dependency and remove eslint
- Change `lint:xxx` scripts to use goodparts binary instead of eslint
- Linter errors should be fixed now
- Hack to get goodparts working in atom:
  - Add `lint:init` script to symlink the goodparts `.eslintrc.js` into root
  - **NOTE:** Must be using the global `eslint` package in the atom `linter-eslint` package settings

### CODE REVIEW

**Comment by samhstn** [comment]
To be merged after #41

**Comment by eliasmalik** [comment]
@jrans, @nelsonic Have a look!

File: package.json [hunk 1]
```diff
@@ -18,16 +18,19 @@
...
     "istanbul": "^0.4.5",
     "pre-commit": "^1.1.3",
     "tape": "^4.6.0"
   },
   "scripts": {
     "test": "node_modules/.bin/tape ./test/*.test.js",
-    "lint": "node_modules/.bin/eslint .",
-    "lint:fix": "node_modules/.bin/eslint . --fix",
+    "lint": "node_modules/.bin/goodparts .",
+    "lint:init": "ln -sf node_modules/goodparts/.eslintrc.js ./.eslintrc.js",
```
Comment (jrans) [inline package.json]: See how lin and lint fix are handy scripts to add but feel like lint:init should be done on the postinstall of the goodparts module. Strange that those goodparts goodtime guys would expect you to add this script to your package.json before installing to get the linting kicked off..

File: package.json [hunk 2]
```diff
@@ -18,16 +18,19 @@
...
+    "lint:fix": "node_modules/.bin/goodparts . --fix",
     "coverage": "node_modules/.bin/istanbul cover node_modules/.bin/tape ./test/*.test.js",
-    "codecov": "node_modules/.bin/codecov --token=${CODECOV_TOKEN}"
+    "codecov": "node_modules/.bin/codecov --token=${CODECOV_TOKEN}",
+    "postinstall": "npm run lint:init"
```
Comment (jrans) [inline package.json]: :fire:

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ node_modules
 coverage
 npm-debug.log
 .eslintrc*
+config.env
```
Comment (jrans) [inline .gitignore:6]: strictly shouldn't be in this PR but will let it slide just this once ;)

**Review by jrans — CHANGES_REQUESTED** [review]
Great work @eliascodes with the rule just one minor change on he package.json

File: package.json (hunk 1 above)
Comment (eliasmalik) [reply package.json]: 😒

**Review by jrans — APPROVED** [review]

**Pull request merged**